### DISCUSSION TITLE
Autohide taskbar only when a window is maximized

### DISCUSSION THREAD

**Discussion by jpedrelli** [ramensoftware/windhawk-mods#1123] [Mod ideas]
A usefull feature like KDE Plasma

**Comment by jpedrelli** [comment]
[SmartTaskbar](https://github.com/Oliviaophia/SmartTaskbar) is a good solution, but don'support dual monitor

**Reply by m417z** [reply]
The taskbar doesn't support per-monitor autohide. Adding support to it will require some work.
For now, here's a simple mod to enable autohide if any window on any monitor is maximized:
https://windhawk.net/mods/taskbar-auto-hide-when-maximized

**Reply by jpedrelli** [reply]
Good! I missed the mod...
If multimonitor support will be available, I will be really happy

**Reply by m417z** [reply]
You didn't miss it, I created the mod after seeing your message :)
I just updated it to version 1.1, and multiple monitors are supported now, and it generally works much better. Check it out and let me know how it works for you.

**Reply by jonasjacob91** [reply]
Autohide per monitor would be really nice. Now I only have the programs that are open on the monitor in the taskbar and the clock is also hidden. My main monitor is 4k, so it does not bother me that the taskbar is visible, but the second is only 1080p. The taskbar takes up too much space

**Comment by silentpig1985** [comment]
![微信图片_20241204165859](https://github.com/user-attachments/assets/b5f65722-26c1-4dfa-84b8-7121f759aa56)
ineffective

**Reply by m417z** [reply]
Did you enable taskbar autohide in Windows taskbar settings?

**Reply by silentpig1985** [reply]
![微信图片_20241207145529](https://github.com/user-attachments/assets/224d9909-eb49-4671-a660-957d78868563)

**Reply by silentpig1985** [reply]
not have

**Reply by explor202** [reply]
I need just the opposite! After finally updating to W11 from 8.1 after all these years, my screensaver (GPhotoShow) appears in W11 as always maximized, so my hidden taskbar won't pop back into view when I hover over it. The DEV suggested adding a custom layout and some pixel padding at the bottom, which sometimes works, but mostly doesn't (when it does work, the taskbar actually appears under the screensaver). My workaround is to keep Notepad open at the extreme bottom right corner so I can always bring my hidden taskbar back into view while the screensaver is running. Is there a way to create a mod that always IGNORES maximized windows and allows the hidden taskbar to be brought back into view?
